test/e2e/specs/sidebar-behaviour.test.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Internal dependencies
+ */
+import '../support/bootstrap';
+import { clearLocalStorage, newPost, newDesktopBrowserPage, setDesktopViewPort, setMobileViewPort } from '../support/utils';
+
+const SIDEBAR_SELECTOR = '.edit-post-sidebar';
+const ACTIVE_SIDEBAR_TAB_SELECTOR = '.edit-post-sidebar__panel-tab.is-active';
+const ACTIVE_SIDEBAR_BUTTON_TEXT = 'Document';
+
+describe( 'Publishing', () => {
+	beforeAll( async () => {
+		await newDesktopBrowserPage();
+	} );
+
+	afterEach( async () => {
+		await clearLocalStorage();
+		await page.goto( 'about:blank' );
+		await setDesktopViewPort();
+	} );
+
+	it( 'Should have sidebar visible at the start with document sidebar active on desktop', async () => {
+		await newPost();
+		const { nodesCount, content, height, width } = await page.$$eval( ACTIVE_SIDEBAR_TAB_SELECTOR, ( nodes ) => {
+			const firstNode = nodes[ 0 ];
+			return {
+				nodesCount: nodes.length,
+				content: firstNode.innerText,
+				height: firstNode.offsetHeight,
+				width: firstNode.offsetWidth,
+			};
+		} );
+
+		// should have only one active sidebar tab.
+		expect( nodesCount ).toBe( 1 );
+
+		// the active sidebar tab should be document.
+		expect( content ).toBe( ACTIVE_SIDEBAR_BUTTON_TEXT );
+
+		// the active sidebar tab should be visible
+		expect( width ).toBeGreaterThan( 10 );
+		expect( height ).toBeGreaterThan( 10 );
+	} );
+
+	it( 'Should have the sidebar closed by default on mobile', async () => {
+		await setMobileViewPort();
+		await newPost();
+		// the page was resized to mobile
+		const width = await page.evaluate(
+			() => {
+				return window.innerWidth;
+			}
+		);
+		expect( width ).toBeLessThan( 782 );
```

Comment:
Aren't we expecting this to be guaranteed by `setMobileViewPort` ?

Reply:
Yes it should be guaranteed, I removed this check.

Reply:
After the removal of this check the test sometimes started to fail. It turns out it was related with puppeteer issue https://github.com/GoogleChrome/puppeteer/issues/1751.

I added a waitForFunction to address it.